### PR TITLE
Add Presence Penalty support to `create_generator`

### PR DESCRIPTION
**Pull request by AirRunner**
`mlx_lm` already implements `make_presence_penalty` in `sample_utils.py`, but `mlx_engine` did not expose it through `create_generator`. This makes it impossible for callers to apply a presence penalty, which is useful for reducing token repetition in long generations.

## Changes

Added `presence_penalty` and `presence_context_size` to `create_generator`, `_sequential_generation`, and `_batched_generation`, following the existing `repetition_penalty` / `repetition_context_size` interface. Both default to `None`.

### Refactor: `TokenPenaltyProcessor`

The existing `RepetitionPenaltyProcessor` was a custom wrapper whose sole purpose was to prepend cached prefix tokens to the penalty window (since logits processors only receive tokens generated in the current turn, not those already in the KV cache). This logic is now extracted into a generic `TokenPenaltyProcessor` that works with any `mlx_lm` penalty function.

## Testing

Added `test_presence_penalty_applies` mirroring the existing `test_repetition_penalty_applies`.

---

This PR exposes `presence_penalty` at the Python level. To make it available in the LM Studio UI, the Node bridge needs to map `presencePenalty` to `presence_penalty` in the kwargs passed to `create_generator` (analogous to how `llm.prediction.llama.presencePenalty` is already handled for the llama.cpp backend).

See [lmstudio-bug-tracker#1604](https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/1604), [lmstudio-bug-tracker#1842](https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/1842).

### CODE REVIEW

**Comment by AirRunner** [comment]
I have read the CLA Document and I hereby sign the CLA

File: mlx_engine/utils/generation_helpers.py [hunk 1]
```diff
@@ -22,38 +22,39 @@
...
-            prompt_tokens[: -len(input_tokens)]
-            if len(input_tokens) > 0
-            else prompt_tokens
+            prompt_tokens[: -len(input_tokens)] if len(input_tokens) > 0 else prompt_tokens
         )
+
+    if repetition_penalty and repetition_penalty != 0.0:
+        logits_processors.append(
+            TokenPenaltyProcessor(
+                make_repetition_penalty(repetition_penalty, repetition_context_size or 20),
```
[the remark this answers is not in the document]
Comment (AirRunner) [reply mlx_engine/utils/generation_helpers.py]: Fixed in 655d788 using an explicit `is not None` check.

(In practice a context size of 0 makes no semantic sense for penalty, but the explicit check is still cleaner).

**Comment by reneleonhardt** [comment]
Thank you very much for your work, everyone is waiting for `presence_penalty` to enjoy the amazing agentic coding results of Qwen 3.6 27b!

Is there a human here to start the workflows and review?

https://unsloth.ai/docs/models/qwen3.6#qwen3.6-27b
https://www.reddit.com/r/LocalLLaMA/comments/1strodp/qwen_36_27b_makes_huge_gains_in_agency_on/

**Comment by AirRunner** [comment]
> Is there a human here?

@reneleonhardt haha what times we live in 😅
